messaging/coap/coap.h

```diff
@@ -180,12 +180,35 @@ size_t coap_serialize_message(void *packet, uint8_t *buffer);
 size_t coap_oscore_serialize_message(void *packet, uint8_t *buffer, bool inner,
                                      bool outer, bool oscore);
 void coap_send_message(oc_message_t *message);
+
+/**
+ * @brief Parse CoAP message options
+ *
+ * @param packet pointer to coap_packet_t struct
+ * @param data raw message data
+ * @param data_len length of raw message data
+ * @param current_option offset of current option in raw message data
+ * @param inner oscore inner option
+ * @param outer oscore outer option
+ * @param oscore oscore used
+ * @param validate if true, it doesn't modify data
+ * @return coap_status_t
+ */
 coap_status_t coap_oscore_parse_options(void *packet, uint8_t *data,
```

Comment:
If I recall correctly, the headers from `messaging/coap` folders are included in public headers. I suspect that's because they are included in other public headers which use some coap constants and/or types. In which case we should reorganize the library and make public only what's necessary from `messaging/coap`. Right now, everything is public, so all functions are public API functions, which is definitely not right. I'll write down an issue.